src/LightGBM.jl

```diff
@@ -39,7 +40,16 @@ end
 
 
 function __init__()
-    LGBM_library[] = Libdl.dlopen(find_library("lib_lightgbm", [@__DIR__]))
+    if Sys.isapple()
+        # for macOS users alternative LightGBM_jll - binaries compiled with Binary Builder
+        # particularly to address the issues raised for M1/aarch64-apple-darwin
+        @eval using LightGBM_jll
+        LGBM_library[] = Libdl.dlopen(find_library("lib_lightgbm", [@__DIR__]))
```

Comment:
```
@eval using LightGBM_jll
```
The `using` can safely be moved to the package level. The source code inside `LightGBM_jll` is fairly minimal and shouldn't really affect startup time. Also, if the dependency is specified in Project, then it will be downloaded already anyway.